### DISCUSSION TITLE
`char-syntax` reports a different syntax than `describe-char`, breaking some functionalities

### DISCUSSION THREAD

**Discussion by agreselin** [emacs-gnuplot/gnuplot#92] [Q&A]
Something in the syntax setup (use of [syntax properties](https://www.gnu.org/software/emacs/manual/html_node/elisp/Syntax-Table-Internals.html)? I don't know much about this matter) causes `(char-syntax (char-after))` to return a different class than `describe-char` or `(syntax-class-to-char (syntax-class (syntax-after (point))))` for some characters, e.g. `#` and `"`. This fools `forward-same-syntax` (it gets stuck at `#` or quotes), `electric-quote-local-mode` and `electric-pair-mode` (the `electric-pair-inhibit-predicate` and `electric-pair-text-syntax-table` don’t work as expected).

**Comment by minad** [comment]
I've seen similar inconsistencies in Elisp modes, `(char-syntax (char-after))` vs `(char-syntax (char-after))`. Could you please describe in more detail what you see and what you expect precisely? I don't observe problems with `electric-pair-mode`. It inserts "" as expected when pressing the " key.
